src/pkg/scan/handler.go

```diff
@@ -44,8 +47,21 @@ type Handler interface {
 	RequiredPermissions() []*types.Policy
 	// RequestParameters defines the parameters for scan request
 	RequestParameters() map[string]interface{}
-	// ReportURLParameter defines the parameters for scan report
-	ReportURLParameter(sr *v1.ScanRequest) (string, error)
 	// PostScan defines the operation after scan
 	PostScan(ctx job.Context, sr *v1.ScanRequest, rp *scan.Report, rawReport string, startTime time.Time, robot *model.Robot) (string, error)
+	ReportHandler
+}
+
+// ReportHandler handler for scan report, it could be sbom report or vulnerability report
+type ReportHandler interface {
```

Comment:
If we define it as the report handler, we could not use the name Report in the methods name. Whenever the caller calls, it sounds like, reportHandler.UpdateData()
reportHandler.GetPlaceHolder()